### PR TITLE
Add --sync flag to `admin bucket remote add` command

### PR DESCRIPTION
**Pull request by poornas**
This flag allows synchronous replication to be specified for a
remote replication target

### CODE REVIEW

**Comment by kannappanr** [comment]
@poornas PTAL build failures. Should we wait for MinIO Pr to be merged?

**Review by kannappanr — APPROVED** [review]
LGTM

File: cmd/admin-bucket-remote-add.go [hunk 1]
```diff
@@ -88,9 +98,10 @@ EXAMPLES:
          --service "replication" --region "us-west-1"
 
   2. Set a new remote replication target 'targetbucket' in region "us-west-1" on https://minio.siteb.example.com for
-     bucket 'sourcebucket' with bandwidth set to 2 gigabits per second.
+	 bucket 'sourcebucket' with bandwidth set to 2 gigabits per second. Enable synchronous replication to the target
+	 and perform health check of target every 100 seconds
      {{.Prompt}} {{.HelpName}} sitea/sourcebucket https://foobar:foo12345@minio.siteb.example.com/targetbucket \
-         --service "replication" --region "us-west-1 --bandwidth "2G"
+         --service "replication" --region "us-west-1 --bandwidth "2G" --sync --healthcheck-seconds 100
```
Comment (harshavardhana) [inline cmd/admin-bucket-remote-add.go:104]:
```suggestion
         --service "replication" --region "us-west-1 --bandwidth "2G" --sync
```

We can leave our healthcheck tunable 60secs is more than enough for most, it is present for most advanced use cases only.

File: docs/minio-admin-complete-guide.md [hunk 2]
```diff
@@ -897,10 +897,10 @@ mc admin bucket remote add myminio/srcbucket https://foobar:foobar12345@minio2:9
 ARN = `arn:minio:replication:us-west-1:1f8712ba-e38f-4429-bcb1-a7bb5aa97447:targetbucket`
 ```
 
-*Example: Add a new replication target `targetbucket` in region `us-west-1` on `https://minio2:9000` for bucket `srcbucket` on MinIO server. `foobar` and `foo12345` are credentials to target endpoint. The max bandwidth is metric 2G (2*10^9)*
+*Example: Add a new replication target `targetbucket` in region `us-west-1` on `https://minio2:9000` for bucket `srcbucket` on MinIO server. `foobar` and `foo12345` are credentials to target endpoint. The max bandwidth is metric 2G (2*10^9). Enable synchronous replication to this target, set healthcheck duration for checking target availability to 100 seconds*
 
 ```
-mc admin bucket remote add myminio/srcbucket https://foobar:foobar12345@minio2:9000/targetbucket --service "replication" --region "us-west-1" --bandwidth "2G"
+mc admin bucket remote add myminio/srcbucket https://foobar:foobar12345@minio2:9000/targetbucket --service "replication" --region "us-west-1" --bandwidth "2G" --sync --healthcheck-seconds 100
```
Comment (harshavardhana) [inline docs/minio-admin-complete-guide.md:903]:
```suggestion
mc admin bucket remote add myminio/srcbucket https://foobar:foobar12345@minio2:9000/targetbucket --service "replication" --region "us-west-1" --bandwidth "2G" --sync
```

Leave out `healthcheck-seconds` as its an advanced usecase, not worthy of prime time in our examples :-)

**Review by harshavardhana — CHANGES_REQUESTED** [review]

File: cmd/admin-bucket-remote-add.go [hunk 3]
```diff
@@ -50,6 +51,15 @@ var adminBucketRemoteAddFlags = []cli.Flag{
 		Name:  "bandwidth",
 		Usage: "Set bandwidth limit in bits per second (K,B,G,T for metric and Ki,Bi,Gi,Ti for IEC units)",
 	},
+	cli.BoolFlag{
+		Name:  "sync",
+		Usage: "enable synchronous replication for this target. Default is async",
+	},
+	cli.UintFlag{
+		Name:  "healthcheck-seconds",
+		Usage: "health check duration in seconds",
```
Comment (vadmeste) [inline cmd/admin-bucket-remote-add.go:60]: What it does exactly ? health check timeout ?

File: cmd/admin-bucket-remote-add.go [hunk 4]
```diff
@@ -198,15 +218,18 @@ func fetchRemoteTarget(cli *cli.Context) (sourceBucket string, bktTarget *madmin
...
+		Credentials:         creds,
+		Endpoint:            u.Host,
+		Path:                path,
+		API:                 "s3v4",
+		Type:                madmin.ServiceType(serviceType),
+		Region:              cli.String("region"),
+		BandwidthLimit:      int64(bandwidth),
+		Label:               strings.ToUpper(cli.String("label")),
+		ReplicationSync:     cli.Bool("sync"),
+		HealthCheckDuration: time.Duration(cli.Uint("healthcheck-seconds")),
```
Comment (vadmeste) [inline cmd/admin-bucket-remote-add.go:232]: I think `* time.Second` is missing here

**Review by vadmeste — CHANGES_REQUESTED** [review]
Some comments

File: cmd/admin-bucket-remote-add.go (hunk 3 above)
Comment (poornas) [reply cmd/admin-bucket-remote-add.go:60]: yes

Comment (vadmeste) [reply cmd/admin-bucket-remote-add.go:60]: > frequency of health check to see if remote target is up

okay, so I guess you need to update this flag usage 'health check duration in seconds' is not that clear. like, 'Health check interval in seconds'

**Review by vadmeste — COMMENTED** [review]
One comment

**Comment by poornas** [comment]
PTAL @harshavardhana ,@vadmeste

**Review by vadmeste — APPROVED** [review]
LGTM

**Pull request merged**